engine/src/test/java/com/datatorrent/stram/engine/OperatorContextTest.java

```diff
@@ -0,0 +1,112 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datatorrent.stram.engine;
+
+import javax.validation.constraints.NotNull;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.apache.hadoop.conf.Configuration;
+
+import com.google.common.base.Preconditions;
+
+import com.datatorrent.api.Attribute;
+import com.datatorrent.api.AutoMetric;
+import com.datatorrent.api.Context;
+import com.datatorrent.api.DAG;
+import com.datatorrent.api.DefaultOutputPort;
+import com.datatorrent.api.InputOperator;
+import com.datatorrent.api.LocalMode;
+import com.datatorrent.api.StreamingApplication;
+import com.datatorrent.common.util.BaseOperator;
+import com.datatorrent.stram.api.OperatorDeployInfo;
+
+public class OperatorContextTest
+{
+  private static boolean foundOperatorName;
+
+  @Test
+  public void testInjectionOfOperatorName() throws Exception
+  {
+    final LocalMode lma = LocalMode.newInstance();
+    StreamingApplication testApp = new StreamingApplication()
+    {
+      @Override
+      public void populateDAG(DAG dag, Configuration conf)
+      {
+        MockInputOperator input = dag.addOperator("input", new MockInputOperator());
+        GenericNodeTest.GenericOperator output = dag.addOperator("output", new GenericNodeTest.GenericOperator());
```

Comment:
@vrozov strictly considering the code for this test case you are technically correct. But I think there is a broader issue here. Apex on the whole has very poor testing of the critical paths. A good example is checkpointing. I don't believe there is a single test which verifies that the logic which determines the checkpoint a node is restored to is correct even though the logic can easily be pulled into a class and tested independently. Cycles should be spent correcting those issues instead of optimizing a trivial test to perfection.

At the end of the day software is written by people and people respond to incentives. If people are given a hard time about small tests, they will be incentivized not to write tests. And so components will remain untested.